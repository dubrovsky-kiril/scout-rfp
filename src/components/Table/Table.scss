.table {
  color: rgba(216, 112, 147, 1);
  border-collapse: collapse;
  width: 100%;
  td {
    border: 2px solid rgba(216, 112, 147, 1);

    padding: 10px;
  }
  thead th {
    border: 2px solid rgba(216, 112, 147, 1);
    padding: 10px;

    font-size: 18px;
    text-align: center;
    font-weight: bold;
    background-color: #fff;
    width: 50%;
  }
  tbody tr > td {
    &:last-child {
      text-align: center;
    }
  }
  tbody tr {
    background-color: #fff;
    font-size: 14px;
    &:hover {
      background-color: rgba(255, 255, 255, 0.8);
      transition: 0.3s;
    }
  }
}
