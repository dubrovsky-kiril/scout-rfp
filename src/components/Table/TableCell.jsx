import React, { memo } from "react";
import PropTypes from "prop-types";

const TableCell = memo(({ cell }) => {
  return <td>{cell}</td>;
});

TableCell.propTypes = {
  cell: PropTypes.string.isRequired
};

export default TableCell;
