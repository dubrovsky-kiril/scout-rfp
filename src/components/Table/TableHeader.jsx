import React, { memo } from "react";
import PropTypes from "prop-types";

const TableHeader = ({ title }) => {
  return <th>{title}</th>;
};

TableHeader.propTypes = {
  title: PropTypes.string.isRequired
};

export default memo(TableHeader);
