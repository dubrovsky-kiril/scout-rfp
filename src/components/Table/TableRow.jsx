import React from "react";
import PropTypes from "prop-types";
import TableCell from "#components/Table/TableCell";

const TableRow = ({ row }) => {
  return (
    <tr>
      {row.map((cell, index) => {
        return <TableCell key={index} cell={cell} />;
      })}
    </tr>
  );
};

TableRow.propTypes = {
  row: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default TableRow;
