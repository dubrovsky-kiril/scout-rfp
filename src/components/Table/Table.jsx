import React from "react";
import PropTypes from "prop-types";
import TableRow from "#components/Table/TableRow";
import TableHeader from "#components/Table/TableHeader";
import styles from "./Table.scss";

const Table = ({ data }) => {
  const columnHeaders = [...data[0]];
  const rows = data.slice(1);

  return (
    <table className={styles.table}>
      <thead>
        <tr>
          {columnHeaders.map((header, index) => {
            return <TableHeader key={index} title={header} />;
          })}
        </tr>
      </thead>
      {rows.length > 0 && (
        <tbody>
          {rows.map((row, index) => {
            return <TableRow key={index} row={row} />;
          })}
        </tbody>
      )}
    </table>
  );
};

Table.propTypes = {
  data: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)).isRequired
};

export default Table;
