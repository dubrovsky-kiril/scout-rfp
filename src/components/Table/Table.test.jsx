import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import Table from "./Table";

describe("Table component", () => {
  test("Matches the snapshot", () => {
    const data = [["X", "Y"], ["A", "B"], ["0", "9"]];
    const table = create(<Table data={data} />);

    expect(table.toJSON()).toMatchSnapshot();
  });
  test("Is showing expected rows number", () => {
    const data = [["X", "Y"], ["A", "B"], ["0", "9"]];
    const rowsNumber = data.length;

    const table = mount(<Table data={data} />);

    expect(table.find("tr").length).toEqual(rowsNumber);
  });
  test("Is showing expected header cells number", () => {
    const data = [["X", "Y"], ["A", "B"], ["0", "9"]];
    const headersCellsNumber = data[0].length;

    const table = mount(<Table data={data} />);

    expect(table.find("th").length).toEqual(headersCellsNumber);
  });
  test("Is showing expected row cells number", () => {
    const data = [["X", "Y"], ["A", "B"], ["0", "9"]];
    const cellsPerRow = data[0].length;

    const table = mount(<Table data={data} />);

    table
      .find("tbody")
      .find("tr")
      .map(node => expect(node.find("td").length).toEqual(cellsPerRow));
  });
  test("Is showing expected header cells values", () => {
    const data = [["X", "Y"], ["A", "B"], ["0", "9"]];
    const headersCells = data[0];

    const table = mount(<Table data={data} />);

    table
      .find("th")
      .map((node, index) => expect(node.text()).toEqual(headersCells[index]));
  });
  test("Is showing expected rows cells values", () => {
    const data = [["X", "Y"], ["A", "B"], ["0", "9"]];
    const rowsCells = data.slice(1);

    const table = mount(<Table data={data} />);

    table
      .find("tbody")
      .find("tr")
      .map((trNode, trIndex) => {
        return trNode
          .find("td")
          .map((tdNode, tdIndex) =>
            expect(tdNode.text()).toEqual(rowsCells[trIndex][tdIndex])
          );
      });
  });
});
