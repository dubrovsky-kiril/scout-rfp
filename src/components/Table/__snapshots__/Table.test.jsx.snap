// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component Matches the snapshot 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        X
      </th>
      <th>
        Y
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        A
      </td>
      <td>
        B
      </td>
    </tr>
    <tr>
      <td>
        0
      </td>
      <td>
        9
      </td>
    </tr>
  </tbody>
</table>
`;
